### PR TITLE
Fix  Refactor Dependencies #10

### PR DESCRIPTION
**Pull request by AnshuTrivedi**
* Removed  contrib dependencies from [tool.poetry.dev-dependencies]
* Add [tool.poetry.extras] and enlisted contrib dependencies

### CODE REVIEW

**Review by AnshuTrivedi — COMMENTED** [review]
@apiad I made changes in dependencies as per your guidance.
I need feed back if there is other changes in files dependent on this issue.

File: pyproject.toml [hunk 1]
```diff
@@ -22,31 +22,35 @@ pydot = "^1.4.1"
...
 mkdocs-material = "^4.6.0"
 mkdocs-pdf-export-plugin = "^0.5.5"
 mypy = "^0.761"
-nltk = "^3.4.5"
 pylint = "^2.4.4"
 pytest = "^5.3.2"
 pytest-cov = "^2.8.1"
 python-telegram-bot = "^12.4.2"
-scikit-learn = "^0.22"
 seqlearn = "^0.2"
```
Comment (apiad) [inline pyproject.toml:37]: :point_up_2:  This dependency is also and extra.

File: pyproject.toml [hunk 2]
```diff
@@ -22,31 +22,35 @@ pydot = "^1.4.1"
...
 sklearn_crfsuite = "^0.3.6"
```
Comment (apiad) [inline pyproject.toml:38]: :point_up_2:  This dependency is also and extra.

File: pyproject.toml [hunk 3]
```diff
@@ -22,31 +22,35 @@ pydot = "^1.4.1"
...
-spacy = "^2.2.3"
-streamlit = "^0.59.0"
 transformers = "^2.3.0"
```
Comment (apiad) [inline pyproject.toml:39]: :point_up_2:  This dependency is also and extra.

File: pyproject.toml [hunk 4]
```diff
@@ -22,31 +22,35 @@ pydot = "^1.4.1"
...
 mkautodoc = "^0.1.0"
 mkdocs = "^1.0.4"
 mkdocs-material = "^4.6.0"
 mkdocs-pdf-export-plugin = "^0.5.5"
 mypy = "^0.761"
-nltk = "^3.4.5"
 pylint = "^2.4.4"
 pytest = "^5.3.2"
 pytest-cov = "^2.8.1"
 python-telegram-bot = "^12.4.2"
```
Comment (apiad) [inline pyproject.toml:36]: :point_up_2:  This dependency is also and extra.

File: pyproject.toml [hunk 5]
```diff
@@ -22,31 +22,35 @@ pydot = "^1.4.1"
 [tool.poetry.dev-dependencies]
 codecov = "^2.0.15"
 cython = "^0.29.14"
-gensim = "^3.8.1"
 jupyter = "^1.0.0"
 jupyterlab = "^1.2.4"
-keras = "^2.3.1"
 matplotlib = "^3.1.2"
```
Comment (apiad) [inline pyproject.toml:27]: This one, I'm not even sure why we need it at all :sweat_smile:
I'll have to check where it's being used.

**Comment by apiad** [comment]
Looks great so far! Thanks :pray: ! I marked a few more lines I believe are also extras. The final step would be to check if the unit tests pass without those dependencies. If that works, then we'll be ready to merge.

**Pull request closed without merging**